llvm/include/llvm/DebugInfo/DWARF/DWARFUnit.h

```diff
@@ -257,6 +259,8 @@ class DWARFUnit {
 
   std::shared_ptr<DWARFUnit> DWO;
 
+  mutable std::recursive_mutex FreeDIEsMutex;
+
```

Comment:
Previously we tried pretty hard to keep the multithreading support isolated in the ThreadSafeDWARFContext or something like that - could you try to do the same now? (move the locking to the ThreadSafeDWARFContext, probably)

(& yeah, please undo the unrelated formatting changes in this file - it's hard to review with them in)